test/e2e/main_test.go

```diff
@@ -3,57 +3,245 @@ package e2e
-	routeBearerToken := flag.String("route-bearer-token", "", "The bearer token to the reporting-operator serviceaccount if use-route-for-reporting-api is set to true")
+	kubeConfigFlag := flag.String("kubeconfig", "", "kube config path, e.g. $HOME/.kube/config")
+	nsPrefixFlag := flag.String("namespace-prefix", "", "The namespace prefix to install the metering resources.")
+	manifestDirFlag := flag.String("deploy-manifests-dir", "../../manifests/deploy", "The absolute/relative path to the metering manifest directory.")
+	cleanupScriptPathFlag := flag.String("cleanup-script-path", "../../hack/run-test-cleanup.sh", "The absolute/relative path to the testing cleanup hack script.")
+	testOutputPathFlag := flag.String("test-output-path", "", "The absolute/relative path that you want to store test logs within.")
+	logLevelFlag := flag.String("log-level", logrus.DebugLevel.String(), "The log level")
 	flag.Parse()
 
+	logger := testhelpers.SetupLogger(*logLevelFlag)
+
 	var err error
-	if testReportingFramework, err = reportingframework.New(*ns, *kubeconfig, *httpsAPI, *useKubeProxyForReportingAPI, *useRouteForReportingAPI, *routeBearerToken, *reportingAPIURL, reportTestOutputDirectory); err != nil {
-		logrus.Fatalf("failed to setup framework: %v\n", err)
+	if df, err = deployframework.New(
+		logger,
+		*nsPrefixFlag,
+		*manifestDirFlag,
+		*kubeConfigFlag,
+		*cleanupScriptPathFlag,
+		*testOutputPathFlag,
+	); err != nil {
+		logger.Fatalf("Failed to create a new deploy framework: %v", err)
 	}
 
 	os.Exit(m.Run())
 }
 
-func TestReportingProducesData(t *testing.T) {
+func TestMultipleInstalls(t *testing.T) {
+	testInstallConfigs := []struct {
+		TargetPods                int
+		Name                      string
+		MeteringOperatorImageRepo string
+		MeteringOperatorImageTag  string
+		MeteringConfigSpec        metering.MeteringConfigSpec
+	}{
+		{
+			Name:                      "HDFSInstall",
+			TargetPods:                defaultTargetPods,
+			MeteringOperatorImageRepo: meteringOperatorImageRepo,
+			MeteringOperatorImageTag:  meteringOperatorImageTag,
+			MeteringConfigSpec: metering.MeteringConfigSpec{
+				LogHelmTemplate: testhelpers.PtrToBool(true),
+				UnsupportedFeatures: &metering.UnsupportedFeaturesConfig{
+					EnableHDFS: testhelpers.PtrToBool(true),
+				},
+				Storage: &metering.StorageConfig{
+					Type: "hive",
+					Hive: &metering.HiveStorageConfig{
+						Type: "hdfs",
+						Hdfs: &metering.HiveHDFSConfig{
+							Namenode: "hdfs-namenode-0.hdfs-namenode:9820",
+						},
+					},
+				},
+				ReportingOperator: &metering.ReportingOperator{
+					Spec: &metering.ReportingOperatorSpec{
+						Resources: &v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("1"),
+								v1.ResourceMemory: resource.MustParse("250Mi"),
+							},
+						},
+						Image: &metering.ImageConfig{
+							Repository: reportingOperatorImageRepo,
```

Comment:
As a follow up, we need to make sure these are always correctly set. This might be a case where we have to force it set Setup() or something like you had done for namespace cause it'll probably be too much to extract out all these sub-configs.

Reply:
Proposed fix: https://github.com/operator-framework/operator-metering/pull/994